### PR TITLE
Adding getTypeName() to functions.php

### PR DESCRIPTION
**Pull request by burzum**
Suggested by @ADmad [here](https://github.com/cakephp/cakephp/pull/11300#discussion_r143363472). We have right now already 5 cases of this in our code and there will be probably more in the future if we stick to the progress on improving exception messages with meaningful information.

### CODE REVIEW

**Comment by ADmad** [comment]
Could be useful in app/plugin code too :)

File: src/basics.php [hunk 1]
```diff
@@ -155,3 +155,16 @@ function loadPHPUnitAliases()
         require_once dirname(__DIR__) . DS . 'tests' . DS . 'phpunit_aliases.php';
     }
 }
+
+if (!function_exists('get_var_type')) {
```
Comment (dereuromark) [inline src/basics.php]: Since when are we using underscore as convention for method names? getVarType() would be the consistent naming IMO.

Comment (dereuromark) [reply src/basics.php]: The "var" sounds odd here... getTypeOrClassName() maybe?

**Comment by dereuromark** [comment]
I would definitely not abbreviate the "var" => "variable".
But I am sure it is not even needed if we can come up with a better name.

**Comment by inoas** [comment]
just getType... tbh.

**Comment by inoas** [comment]
http://php.net/manual/de/function.gettype.php#104224 do we cover all of these? Including null?

**Comment by dereuromark** [comment]
@ionas Isn't this used inside the Debugger class? And shouldn't this then rather be bound to that scope (as class method `Debugger::getType()`), as this is mainly a debugging convenience method?

**Comment by dereuromark** [comment]
You also have to consider if all subsplits will still work with this method being added to the main functions file.
So far those might have worked without it, and the dependencies should be double-checked.

File: src/basics.php [hunk 2]
```diff
@@ -155,3 +155,16 @@ function loadPHPUnitAliases()
         require_once dirname(__DIR__) . DS . 'tests' . DS . 'phpunit_aliases.php';
     }
 }
+
+if (!function_exists('getVarType')) {
```
Comment (markstory) [inline src/basics.php]: If we intend to use this in classes that end up in sub split packages,this function should be in src/Core/functions.php instead.

**Review by markstory — COMMENTED** [review]
I like the idea and implementation, but the function will likely need to be moved.

**Comment by ADmad** [comment]
@markstory I have moved the function to `functions.php`.

**Review by ADmad — APPROVED** [review]

**Comment by dereuromark** [comment]
The "var" name thing is still wrong IMO. We never add var to anything, and of course the input is usually the object the method operates on.
Best to remove it and stick to `getType()` or `getTypeOrClassName()` then only.

**Comment by ADmad** [comment]
getTypeName() works for me too.

File: src/Core/functions.php [hunk 3]
```diff
@@ -280,3 +280,16 @@ function deprecationWarning($message, $stackFrame = 2)
         trigger_error($message, E_USER_DEPRECATED);
     }
 }
+
+if (!function_exists('getVarType')) {
```
Comment (markstory) [inline src/Core/functions.php]: This needs to reflect the rename as well.

**Pull request merged**